### PR TITLE
chore!: refactor NewMsgPayForBlob to accept blobs

### PR DESCRIPTION
**Pull request by evan-forbes**
## Overview

part of #388 and split off from #1154. This is just a minor refactor for `NewMsgPayForBlob` to accept a `tmproto.Blob`, instead of the individual components.

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

**Review by cmwaters — CHANGES_REQUESTED** [review]

File: x/blob/types/payforblob.go [hunk 1]
```diff
@@ -25,19 +26,19 @@ const (
 
 var _ sdk.Msg = &MsgPayForBlob{}
 
-func NewMsgPayForBlob(signer string, nid namespace.ID, blob []byte) (*MsgPayForBlob, error) {
-	commitment, err := CreateCommitment(nid, blob, appconsts.ShareVersionZero)
+func NewMsgPayForBlob(signer string, blob *tmproto.Blob) (*MsgPayForBlob, error) {
```
[the remark this answers is not in the document]
Comment (evan-forbes) [reply x/blob/types/payforblob.go]: > Now users need to import the tendermint repo as well which they might not have been doing. 

true! would a referenced type make this any better? added in 9dfae2502792e80540a67e4d150e50dc19656637
```go
type Blob = tmproto.Blob
```

> Blob also has the share version which users might be confused as to whether it's necessary or where to get it.

If the `NewBlob` function is used, do you think this is more palatable? There we add the default share version, at least until we have multiple share versions. I have also added a `DefaultShareVersion` constant with docs that specifically say to use that if someone is not sure which one to use. 9dfae2502792e80540a67e4d150e50dc19656637

Those are fair criticisms, but the alternative for multiple blobs is 
```go
func NewMsgPayForBlob(signer string, namespaces [][]byte, data [][]byte, shareversions []uint32)
```
and that feels really clunky to me

**Comment by evan-forbes** [comment]
Do we also want to refactor https://github.com/celestiaorg/celestia-app/blob/f4f6a3adba8c1d31de13ab72d2b2d3ee522ba462/x/blob/types/payforblob.go#L143-L157
in a similar way?

File: pkg/appconsts/appconsts.go [hunk 2]
```diff
@@ -27,6 +27,10 @@ const (
 	// ShareVersionZero is the first share version format.
 	ShareVersionZero = uint8(0)
 
+	// DefaultShareVersion is the defacto share version. Use this if you are
+	// unsure of which version to use.
+	DefaultShareVersion = ShareVersionZero
```
Comment (rootulp) [inline pkg/appconsts/appconsts.go:32]: 👍

**Review by rootulp — DISMISSED** [review]

File: x/blob/types/payforblob.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (cmwaters) [reply x/blob/types/payforblob.go]: > and that feels really clunky to me

Yeah I agree. I saw that in your other PR and was like meh. 

Yeah I like the idea

```go
type Blob = tmproto.Blob

func NewMsgPayForBlobs(signer string, blobs Blob...)

func NewBlob(namespace, data []byte) // uses default

// and together
msg := NewMsgPayForBlobs("callum", NewBlob("chat", "gm"), NewBlob("chat", "gn"))
```

**Pull request merged**